include/device.h

```diff
@@ -172,6 +172,37 @@ extern "C" {
 			pm_control_fn,					\
 			data_ptr, cfg_ptr, level, prio, api_ptr)
 
+/**
+ * @def DEVICE_DT_INST_DEFINE
+ *
+ * @brief Like DEVICE_DT_DEFINE for an instance of a DT_DRV_COMPAT compatible
+ *
+ * @param inst instance number
+ *
+ * @param init_fn Address to the init function of the driver.
+ *
+ * @param pm_control_fn Pointer to device_pm_control function.
+ * Can be empty function (device_pm_control_nop) if not implemented.
+ *
+ * @param data_ptr Pointer to the device's private data.
+ *
+ * @param cfg_ptr The address to the structure containing the
+ * configuration information for this instance of the driver.
+ *
+ * @param level The initialization level.  See SYS_INIT() for
+ * details.
+ *
+ * @param prio Priority within the selected initialization level. See
+ * SYS_INIT() for details.
+ *
+ * @param api_ptr Provides an initial pointer to the API function struct
+ * used by the driver. Can be NULL.
+ */
+#define DEVICE_DT_INST_DEFINE(inst, init_fn, pm_control_fn, data_ptr,	\
```

Comment:
Could this be:
```
#define DEVICE_DT_INST_DEFINE(inst, ...) DEVICE_DT_DEFINE(DT_DRV_INST(inst), __VAR_ARGS__)
```
so we don't have to deal with all the duplicated descriptions of parameters?

Reply:
How would we document that VA_ARGS portion?

Reply:
```
/**
 * @def DEVICE_DT_INST_DEFINE
 *
 * @brief Like DEVICE_DT_DEFINE for an instance of a DT_DRV_COMPAT compatible
 *
 * @param inst instance number.  This is replaced by
 * `DT_DRV_COMPAT(inst)` in the call to DEVICE_DT_DEFINE.
 *
 * @param ...args other parameters as expected by DEVICE_DT_DEFINE.
 */
#define DEVICE_DT_INST_DEFINE(inst, ...) \
	DEVICE_DT_DEFINE(DT_DRV_INST(inst), __VAR_ARGS__)
```

As I think about it, this is not a minor concern.  Part of #29644 involves changing `DEVICE_DT_DEFINE()` to accept trailing arguments that reflect device-specific dependencies that are not derived from devicetree information.

I really don't think we're gaining enough by these wrappers to compensate for the pain it will be to deal with future refactoring.